jetstream/test/jetstream_test.go

```diff
@@ -431,6 +431,236 @@ func TestCreateStreamMirrorCrossDomains(t *testing.T) {
 	}
 }
 
+func TestCreateOrUpdate_CreateStream(t *testing.T) {
+	tests := []struct {
+		name      string
+		stream    string
+		subject   string
+		timeout   time.Duration
+		withError error
+	}{
+		{
+			name:    "create stream, ok",
+			stream:  "foo",
+			timeout: 10 * time.Second,
+			subject: "FOO.1",
+		},
+		{
+			name:    "with empty context",
+			stream:  "foo-o",
+			subject: "FOO.12",
+		},
+		{
+			name:      "invalid stream name",
+			stream:    "foo.123",
+			subject:   "FOO.123",
+			timeout:   10 * time.Second,
+			withError: jetstream.ErrInvalidStreamName,
+		},
+		{
+			name:      "stream name required",
+			stream:    "",
+			subject:   "FOO.1234",
+			timeout:   10 * time.Second,
+			withError: jetstream.ErrStreamNameRequired,
+		},
+		{
+			name:    "update stream, ok",
+			stream:  "foo",
+			subject: "BAR.123",
+			timeout: 10 * time.Second,
+		},
+		{
+			name:      "context timeout",
+			stream:    "foo",
+			subject:   "BAR.1234",
+			timeout:   1 * time.Microsecond,
+			withError: context.DeadlineExceeded,
+		},
+	}
+
+	srv := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, srv)
+	nc, err := nats.Connect(srv.ClientURL())
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	js, err := jetstream.New(nc)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	defer nc.Close()
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx := context.Background()
+			if test.timeout > 0 {
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(context.Background(), test.timeout)
+				defer cancel()
+			}
+			_, err = js.CreateOrUpdateStream(ctx, jetstream.StreamConfig{Name: test.stream, Subjects: []string{test.subject}})
+			if test.withError != nil {
+				if !errors.Is(err, test.withError) {
+					t.Fatalf("Expected error: %v; got: %v", test.withError, err)
+				}
+				return
+			}
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+		})
+	}
+}
+
+func TestCreateOrUpdateStream_CreateMirrorCrossDomains(t *testing.T) {
```

Comment:
I don't think it's useful to have this test for `CreateOrUpdateStream` to be honest, we already test the same functionality in `TestCreateStreamMirrorCrossDomains` so it brings no value. Please remove it.

Reply:
ok. this test has been removed